### PR TITLE
Use Django Amber to generate static site

### PR DESCRIPTION
**Pull request by inglesp**
This PR changes the static site generator we use to build the site, from [wok](http://github.com/mythmon/wok) to [Django Amber](https://github.com/inglesp/django-amber/), which is something I've been hacking on recently.

The main benefit to this is that, because Django Amber is built on Django, we don't need to jump through any of wok's hoops to do anything slightly nonstandard.  In particular, we can use our own data model (see models.py) which gives us a lot more flexibility.  For instance, it'll make having multiple versions of the schedule (as we did last year) much easier.

It'll also be easier to do things like getting SASS compilation working.

(It also means that the site runs on Python 3!)

The main drawback is the Django Amber is new and is not very battle-hardened.

Thoughts?

### CODE REVIEW

File: pre-flight-checks.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ ERRORS=()
 
 echo " *** Checking that site can be built."
 
-wok --serve >/dev/null 2>&1 &
-WOK_PID=$!
+python manage.py serve >/dev/null 2>&1 &
```
Comment (ghickman) [inline pre-flight-checks.sh:11]: Now that we've got Django under the hood (pun, so very much intended ;) ) - we could also use the checks framework as an initial test here.

File: pyconuk/settings.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'secret'
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
```
Comment (ghickman) [inline pyconuk/settings.py:26]: Can we pick this up from the environment please?

`DEBUG = bool(int(os.environ.get('DEBUG', False)))`

File: pyconuk/settings.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+
+    'django.contrib.staticfiles',
+    
+    'django_amber',
+    'markdown_deux',
+
+    'pyconuk',
+]
```
Comment (ghickman) [inline pyconuk/settings.py:43]: Apps should be ordered:
1. Project
2. Third Party
3. Django

The way templates are looked up is first come first served based on this ordering so to have app level overrides they must come first.

File: pyconuk/settings.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+LANGUAGE_CODE = 'en-gb'
+
+TIME_ZONE = 'UTC'
+
+USE_I18N = True
+
+USE_L10N = True
+
+USE_TZ = True
```
Comment (ghickman) [inline pyconuk/settings.py:87]: I don't think we need any of these turned on?

File: pre-flight-checks.sh (hunk 1 above)
Comment (inglesp) [inline pre-flight-checks.sh:11]: Yes we could... it's not a priority for me right now though (although it'd be a good excuse to learn my way around the checks framework).

File: pyconuk/settings.py (hunk 2 above)
Comment (inglesp) [inline pyconuk/settings.py:26]: Given that the application never runs in production (instead, to build the static site, we use `runserver` and crawl the pages that get served) I don't see the need to change this.

File: pyconuk/settings.py (hunk 3 above)
Comment (inglesp) [inline pyconuk/settings.py:43]: I learnt something new today, thanks!

File: pyconuk/settings.py (hunk 4 above)
Comment (inglesp) [inline pyconuk/settings.py:87]: We will be adding talk slots to the data model, so I should think having `USE_TZ` switched on will be useful.

I'll remove the `USE_\w\d\d\w` settings though.

**Comment by meatballs** [comment]
```requirements.in``` and ```requirements.txt``` look like they need updating to replace tox with django-amber

File: pyconuk/settings.py (hunk 2 above)
Comment (ghickman) [inline pyconuk/settings.py:26]: Oh good point! That'll take some getting used to, but yes happy to have this left as is.

**Pull request merged**